// @react-stately/menu/dist/main.js
var $cAWze$reactstatelyoverlays = require("@react-stately/overlays");
var $cAWze$react = require("react");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $f39cdb649cd48930$exports = {};

$parcel$export($f39cdb649cd48930$exports, "useMenuTriggerState", () => $f39cdb649cd48930$export$79fefeb1c2091ac3);


function $f39cdb649cd48930$export$79fefeb1c2091ac3(props) {
    let overlayTriggerState = $cAWze$reactstatelyoverlays.useOverlayTriggerState(props);
    let [focusStrategy1, setFocusStrategy] = $cAWze$react.useState(null);
    return {
        focusStrategy: focusStrategy1,
        ...overlayTriggerState,
        open (focusStrategy = null) {
            setFocusStrategy(focusStrategy);
            overlayTriggerState.open();
        },
        toggle (focusStrategy = null) {
            setFocusStrategy(focusStrategy);
            overlayTriggerState.toggle();
        }
    };
}


$parcel$exportWildcard(module.exports, $f39cdb649cd48930$exports);
