// @react-stately/menu/dist/module.js
import {useOverlayTriggerState as $9Xvoh$useOverlayTriggerState} from "@react-stately/overlays";
import {useState as $9Xvoh$useState} from "react";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $a28c903ee9ad8dc5$exports = {};

$parcel$export($a28c903ee9ad8dc5$exports, "useMenuTriggerState", () => $a28c903ee9ad8dc5$export$79fefeb1c2091ac3);


function $a28c903ee9ad8dc5$export$79fefeb1c2091ac3(props) {
    let overlayTriggerState = $9Xvoh$useOverlayTriggerState(props);
    let [focusStrategy1, setFocusStrategy] = $9Xvoh$useState(null);
    return {
        focusStrategy: focusStrategy1,
        ...overlayTriggerState,
        open (focusStrategy = null) {
            setFocusStrategy(focusStrategy);
            overlayTriggerState.open();
        },
        toggle (focusStrategy = null) {
            setFocusStrategy(focusStrategy);
            overlayTriggerState.toggle();
        }
    };
}




export {$a28c903ee9ad8dc5$export$79fefeb1c2091ac3 as useMenuTriggerState};
