// @react-stately/radio/dist/module.js
import {useControlledState as $fQ2SF$useControlledState} from "@react-stately/utils";
import {useMemo as $fQ2SF$useMemo, useState as $fQ2SF$useState} from "react";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $a54cdc5c1942b639$exports = {};

$parcel$export($a54cdc5c1942b639$exports, "useRadioGroupState", () => $a54cdc5c1942b639$export$bca9d026f8e704eb);


let $a54cdc5c1942b639$var$instance = Math.round(Math.random() * 10000000000);
let $a54cdc5c1942b639$var$i = 0;
function $a54cdc5c1942b639$export$bca9d026f8e704eb(props) {
    // Preserved here for backward compatibility. React Aria now generates the name instead of stately.
    let name = $fQ2SF$useMemo(()=>props.name || `radio-group-${$a54cdc5c1942b639$var$instance}-${++$a54cdc5c1942b639$var$i}`
    , [
        props.name
    ]);
    let [selectedValue, setSelected] = $fQ2SF$useControlledState(props.value, props.defaultValue, props.onChange);
    let [lastFocusedValue, setLastFocusedValue] = $fQ2SF$useState(null);
    let setSelectedValue = (value)=>{
        if (!props.isReadOnly && !props.isDisabled) setSelected(value);
    };
    return {
        name: name,
        selectedValue: selectedValue,
        setSelectedValue: setSelectedValue,
        lastFocusedValue: lastFocusedValue,
        setLastFocusedValue: setLastFocusedValue,
        isDisabled: props.isDisabled || false,
        isReadOnly: props.isReadOnly || false
    };
}




export {$a54cdc5c1942b639$export$bca9d026f8e704eb as useRadioGroupState};
